database/migrations/2022_01_28_104214_create_questionnaire_table.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+use EscolaLms\Core\Models\User;
+use EscolaLms\Questionnaire\Models\Question;
+use EscolaLms\Questionnaire\Models\Questionnaire;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreateQuestionnaireTable extends Migration
+{
+    public function up(): void
+    {
+        if (!Schema::hasTable('questionnaires')) {
+            Schema::create(
+                'questionnaires',
+                function (Blueprint $table) {
+                    $table->id('id');
+                    $table->string('title');
+                    $table->string('model');
+                    $table->integer('model_id');
```

Comment:
one the same `questionnare` could be attached to many models 

to is q 1:n model relation 

`questionnare` can live without relation 

The workflow is 
1. you create `questionnare` 
2. than attach to models

Reply:
Also this is polimorphic relation https://laravel.com/docs/8.x/eloquent-relationships#one-to-one-polymorphic-relations it's not how you handle those 

it would be something like 

```
    modelable_id - integer
    modelable_type - string
```

Reply:
I would probably declare this as `$table->morph('rateable')` (because 'questionnairable' is too much of a mouthful)

Reply:
Done - in next branch i will add action to add models to questonnaire